### ISSUE TITLE
low precision training upcoming feature tracker

### ISSUE THREAD

**Issue by vkuzo** [pytorch/ao#556]
This is a running list of planned features for low precision training.  As features are completed we plan to delete them from this list, to keep things simple.

# MX

## pytorch/pytorch

* [in progress] improving torch.compile performance for to_mx cast across dim1: https://github.com/pytorch/pytorch/issues/149982

## pytorch/torchao

* [in progress] performance tracker: https://github.com/pytorch/ao/issues/1768

# float8

## performance

* [in progress] optimize torch.compile performance for float8 tensorwise scaling/casting kernels
  * [fixed behind a flag, off by default] https://github.com/pytorch/pytorch/issues/130015
  * [planned] https://github.com/pytorch/pytorch/issues/128063
* [in progress] ensure that float8 rowwise scaling is performant with TP and async TP https://github.com/pytorch/pytorch/issues/149990

## distributed

* [planned] verify integration with PP

## new features

* [planned] MoE support: https://github.com/pytorch/ao/issues/2147
* [planned] float8 SDPA (priority TBD)

## ecosystem

* [in progress] enable TP in torchtune's float8 integration: https://github.com/pytorch/torchtune/issues/2682

## other

* [planned] expose float8 training via the quantize_ API
* [planned] migrate `torchao.float8` code to `torchao.quantization` for better unification with the rest of torchao, in a BC-preserving way: https://github.com/pytorch/ao/issues/894

**Comment by airMeng** [comment]
Has SDPA supported FP8 already like flash-attention https://github.com/Dao-AILab/flash-attention/blob/main/hopper/benchmark_flash_attention_fp8.py does?

**Comment by vkuzo** [comment]
> Has SDPA supported FP8 already like flash-attention https://github.com/Dao-AILab/flash-attention/blob/main/hopper/benchmark_flash_attention_fp8.py does?

we would be interesting in accepting a contribution for this!  To me, the priority is not clear as we have not seen convincing evidence that SDPA + float8 can achieve good accuracy numerically.

**Comment by avizon-aws** [comment]
@vkuzo , is there a plan for MXFP8 all-gather? If so, by when is the feature expected to be enabled?

**Comment by vkuzo** [comment]
> [@vkuzo](https://github.com/vkuzo) , is there a plan for MXFP8 all-gather? If so, by when is the feature expected to be enabled?

we have considered this for float8 and mx, but it has not made the priority cut yet.  Could you share some context about your use case, and whether you need this for FSDP or TP?
